src/RF24Log.h

```diff
@@ -0,0 +1,274 @@
+    StreamType* handler; /** the output stream */
+    char _name[40]; /** the logger instance's name (max of 39 characters)*/
+
+public:
+
+    /** Empty constructor. level defaults to @ref NOT_SET, and instance has no handler. */
+    RF24Logger() : level(NOT_SET), handler(nullptr)
+    {
+        _name[0] = 0;
+    }
+
+    /**
+     * Instance constructor. The , and the instance's
+     * handler is initialized.
+     * @param stream The handler to which all logging output will be directed.
+     * @param name The origin's name of the logger's messages.
+     * @param lvl The instance's logging level defaults to @ref NOT_SET if not specified.
+     */
+    RF24Logger(StreamType* stream, const char* name, uin8_t lvl = NOT_SET) : level(lvl), handler(stream)
+    {
+        strcpy(_name, name);
+    }
+
+    /**
+     * Copy constructor. Instance's log level and handler are set to @p obj instance's
+     * corresponding values.
+     * @param obj An instantiated RF24Logger object from which values are copied from.
+     */
+    RF24Logger(RF24Logger* obj) : level(obj->level), handler(obj->handler)
+    {
+        _name[0] = 0;
+    }
+
+    /**
+     * Set the handler to which all logging messages are directed.
+     * @param stream The output stream to be used as the handler.
+     */
+    void setHandler(StreamType* stream)
+    {
+        handler = stream;
+    }
+
+    /**
+     * Set the logging level that's to filter logging messages passed to log()
+     * @param lvl The logging level must be in range [0, 60).
+     */
+    void setLevel(uint8_t lvl)
+    {
+        level = rf24_min(lvl, ERROR + 9);
+    }
+
+    /**
+     * @brief Set a default @p name for the instance
+     *
+     * @param name This will be used in calls to log(uint8_t lvl, Ts msg...)
+     */
+    void setName(const char* name)
+    {
+        strcpy(_name, name);
+    }
+
+    /**
+     * Returns the logging handler that was configured with RF24Logger(RF24Logger*),
+     * RF24Logger(StreamType*, const char*) or setHandler()
+     *
+     * Example usage:
+     * @code
+     * logging.getLogger() << "a string of text " << 0 << ':' << 1.0 << endl;
+     * @endcode
+     * would ouput "a string of text 0:1.0" with a trailing line feed.
+     * @returns The instantiated output stream object passed to setHandler()
+     */
+    StreamType &getLogger()
+    {
+        return *handler;
+    }
+
+    /**
+     * Log a message
+     * @param lvl The logging level used for the specified message.
+     * @param msg The specified message.
+     */
+    template <typename... Ts>
+    void log(uint8_t lvl, Ts... msg)
+    {
+        logOrigin(lvl, _name, msg...);
+    }
+
+    /**
+     * @brief Log a message
+     * @param lvl The logging level used for the specified message.
+     * @param origin The orgin of the message.
+     * @param msg The specified message.
+     */
+    template <typename St, typename... Ts>
+    void logOrigin(uint8_t lvl, St origin, Ts... msg)
```

Comment:
I'm sorry, but using sublevels is non-negotiable to me. SLF4J is a "framework" (thus the "F") which is meant to be middle abstraction layer between lib developers and end-users' applications where end-users can implement their own logging mechanism on top of SLF4J. 

Since this PR doesn't prove why sublevels are useful, I pose this thought experiment:
- RF24Network has 4 types of different debug messages (I perceive them as 4 DEBUG sublevels). Actually there's 5 if you include the [generic `SERIAL_DEBUG`](https://github.com/nRF24/RF24Network/blob/4743ce912d9144733b8604e633fc0f8ff8195071/RF24Network_config.h#L54) macro (which I perceive as the parent DEBUG level). [they are defined](https://github.com/nRF24/RF24Network/blob/4743ce912d9144733b8604e633fc0f8ff8195071/RF24Network_config.h#L96-L119) as:
   ```cpp
    #if defined (SERIAL_DEBUG_MINIMAL)
      #define IF_SERIAL_DEBUG_MINIMAL(x) ({x;})
    #else
      #define IF_SERIAL_DEBUG_MINIMAL(x)
    #endif

    #if defined (SERIAL_DEBUG_FRAGMENTATION)
      #define IF_SERIAL_DEBUG_FRAGMENTATION(x) ({x;})
    #else
      #define IF_SERIAL_DEBUG_FRAGMENTATION(x)
    #endif

    #if defined (SERIAL_DEBUG_FRAGMENTATION_L2)
      #define IF_SERIAL_DEBUG_FRAGMENTATION_L2(x) ({x;})
    #else
      #define IF_SERIAL_DEBUG_FRAGMENTATION_L2(x)
    #endif

    #if defined (SERIAL_DEBUG_ROUTING)
      #define IF_SERIAL_DEBUG_ROUTING(x) ({x;})
    #else
      #define IF_SERIAL_DEBUG_ROUTING(x)
    #endif
   ```
   in which filtering is currently done by only defining a certain level. Without using sublevels, all of these messages would be output. If you're trying to suggest only filtering messages by origin, how would that apply to this thought experiment?

Reply:
Now I see a real life scenario. With my only four levels that would be impossible: all of these messages will go under ```logger.debug()```.

@2bndy5, I will reconsider then the usage of sublevels.